test/integration/provider/admin/backend_apis_controller_test.rb

```diff
@@ -11,11 +13,13 @@ class Provider::Admin::BackendApisControllerTest < ActionDispatch::IntegrationTe
 
   attr_reader :provider
 
-  test '#index' do
+  test '#index does not list deleted backend apis' do
+    FactoryBot.create(:backend_api, account: @provider, state: BackendApi::DELETED_STATE)
     get provider_admin_backend_apis_path
     assert_response :success
-    assert_select '#backend_apis tr', count: @provider.backend_apis.count+1
-    @provider.backend_apis.each do |backend_api|
+    expected_backend_apis = 3 # 1 default + 2 created in setup
```

Comment:
I think it is a safer approach to do: `expected_backend_apis = BackendApi.count - 1`
being the `-1` the one you just created.
This way we don't rely on having 1 or 0 on the default which is something that can change as we have seen and shouldn't be related to this test.